router/src/lib.rs

```diff
@@ -1604,4 +1638,40 @@ mod tests {
             r#"{"role":"assistant","tool_calls":[{"id":"0","type":"function","function":{"description":null,"name":"myfn","arguments":{"format":"csv"}}}]}"#
         );
     }
+
+    #[test]
+    fn tool_choice_formats() {
+        #[derive(Deserialize)]
+        struct TestRequest {
+            tool_choice: ChatCompletionToolChoiceOption,
+        }
+
+        let none = r#"{"tool_choice":"none"}"#;
+        let de_none: TestRequest = serde_json::from_str(none).unwrap();
+        assert_eq!(de_none.tool_choice, ChatCompletionToolChoiceOption::NoTool);
+
+        let auto = r#"{"tool_choice":"auto"}"#;
+        let de_auto: TestRequest = serde_json::from_str(auto).unwrap();
+        assert_eq!(de_auto.tool_choice, ChatCompletionToolChoiceOption::Auto);
+
+        let auto = r#"{"tool_choice":"required"}"#;
+        let de_auto: TestRequest = serde_json::from_str(auto).unwrap();
+        assert_eq!(
+            de_auto.tool_choice,
+            ChatCompletionToolChoiceOption::Required
+        );
+
+        let ref_choice = ChatCompletionToolChoiceOption::Function(FunctionName {
+            name: "myfn".to_string(),
+        });
+
+        let named = r#"{"tool_choice":"myfn"}"#;
+        let de_named: TestRequest = serde_json::from_str(named).unwrap();
+        assert_eq!(de_named.tool_choice, ref_choice);
```

Comment:
```suggestion
        assert_eq!(de_named.tool_choice,ChatCompletionToolChoiceOption::Function(FunctionName {
            name: "myfn".to_string(),
        }) );
```

Please avoid indirection in tests. Tests are about being as dumb as possible, so a reader can easily assess if the test is correct, or outdated. Every indirection makes that harder to reason about.

Reply:
great points, indirection removed in the latest commit